gcovr/html_generator.py

```diff
@@ -382,13 +383,16 @@ def print_html_report(covdata, output_file, options):
         data['source_lines'] = []
         currdir = os.getcwd()
         os.chdir(options.root_dir)
-        with io.open(data['filename'], 'r', encoding=options.source_encoding,
-                     errors='replace') as source_file:
-            lines = formatter.highlighter_for_file(data['filename'])(source_file.read())
-            for ctr, line in enumerate(lines, 1):
-                data['source_lines'].append(
-                    source_row(ctr, line, cdata.lines.get(ctr))
-                )
+        try:
+            with io.open(data['filename'], 'r', encoding=options.source_encoding,
+                         errors='replace') as source_file:
+                lines = formatter.highlighter_for_file(data['filename'])(source_file.read())
+                for ctr, line in enumerate(lines, 1):
+                    data['source_lines'].append(
+                        source_row(ctr, line, cdata.lines.get(ctr))
+                    )
+        except IOError as e:
+            logger.warn('!!! File not found: {filename}, reason={reason}', filename=data['filename'], reason=repr(e))
```

Comment:
The message should only contain the text withou exlamation marks because the logge prepends text.
```suggestion
            logger.warn('File {filename} not found: {reason}', filename=data['filename'], reason=repr(e))
```

Also shall a flag be set to remember the error. This flag must be returned by the function and collected in https://github.com/gcovr/gcovr/blob/4240ddcd212276a6b03a00586fdc859024f66391/gcovr/__main__.py#L438 to set the exit code for a generator error.

Reply:
Generators currently don't report errors. I change this behavior to try to fit your suggestion. Please check for my latest commits. Thanks!